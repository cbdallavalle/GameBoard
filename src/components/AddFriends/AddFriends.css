.AddFriends {
  margin-top: 25px;
}

.AddFriends article {
  width: 600px;
  height: 400px;
  background-color: #ffffffcf;
  overflow-y: scroll;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  margin: auto;
  font-family: 'Open Sans', sans-serif;
}

.AddFriends h4 {
  padding: 25px 10px;
  cursor: pointer;
  text-align: left;
  font-family: 'Open Sans', sans-serif;
}

.display-friends {
  position: relative;
  display: flex;
  justify-content: center;
  width: 635px;
  height: 441px;
  background-color: #f4f4f4a6;
  margin-top: 50px;
}