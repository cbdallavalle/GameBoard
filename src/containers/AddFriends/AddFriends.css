.AddFriends {
  width: 700px;
  margin-top: 25px;
}

.AddFriends article {
  width: 600px;
  height: 400px;
  background-color: #ffffffcf;
  overflow-y: scroll;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  margin: auto;
  font-family: 'Open Sans', sans-serif;
}

.AddFriends h4 {
  padding: 25px 10px;
  cursor: pointer;
  text-align: left;
  font-family: 'Open Sans', sans-serif;
}

.AddFriends h4:hover {
  background-color: #8080804a;
}

#add-friends {
  width: 15px;
  display: inline-block;
  padding-right: 15px;
}

.display-friends {
  position: relative;
  display: flex;
  justify-content: center;
  width: 635px;
  height: 441px;
  background-color: #f4f4f4a6;
  margin: 30px auto;
}