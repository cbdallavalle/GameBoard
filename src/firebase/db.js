import { db } from './firebase';

export const doCreateUser = (id, firstName, lastName, email) =>
  db.ref(`users/${id}`).set({
    user: {
      id,
      firstName,
      lastName,
      email,
    }
  });

export const onceGetUsers = async () => {
  try {
    const snapshot = await db.ref('users').once('value');
    if (snapshot.status < 300) {
      const value = await snapshot.val();
      return value
    } else {
      throw new Error('unable to fetch users')
    }
  } catch (error) {
    throw new Error('unable to fetch users')
  }
}

export const doWriteFavoriteData = async (userId, newFavorites) => {
  try {
    db.ref('users/' + userId + '/favorites').set(newFavorites);
  } catch (error) {
    throw new Error('unable to write favorite data')
  }
}

export const doAddFavoriteData = async (userId, favorite) => {
  try {
    const existingFavorites = await getFavorites(userId);
    const favoriteToAdd = checkFavoriteDuplication(existingFavorites, favorite);
    doWriteLastFavoritedData(userId, favorite)
    const newFavorites = {...existingFavorites, ...favoriteToAdd }
    doWriteFavoriteData(userId, newFavorites)
  } catch (error) {
    throw new Error('unable to add favorite data')
  }
}

export const doDeleteFavoriteData = async (userId, favorite) => {
  try {
    const existingFavorites = await getFavorites(userId);
    delete existingFavorites[favorite.name]
    doWriteFavoriteData(userId, existingFavorites)
  } catch (error) {
    throw new Error('unable to delete')
  }
}

export const doWriteLastFavoritedData = async (userId, favorite) => {
  db.ref('users/' + userId + '/lastFavorited').set(favorite);
}

export const checkFavoriteDuplication = (existingFavorites, favorite) => {
  const { name, id, description, image, thumbnail } = favorite;
  const review = "No review - click to add one";
  const favoriteToAdd = { [name]: { id, description, image, thumbnail, name, review } }
  return existingFavorites && existingFavorites[favorite.name] ? {} : favoriteToAdd
}

export const doWriteFriendsData = async (userId, friendId) => {
  try {
    const friends = await getFriends(userId);
    if (friends && !friends.includes(friendId)) {
      friends.push(friendId);
      db.ref('users/' + userId + '/friends').set(friends); 
    } else if (!friends) {
      db.ref('users/' + userId + '/friends').set([friendId]); 
    }
  } catch (error) {
    throw new Error ('unable to add friend')
  }
}

export const getFriends = async (userId) => {
  try {
    const value = await onceGetUsers();
    return value[userId].friends
  } catch (error) {
    throw new Error('unable to load friends data')
  }
}

export const getFavorites = async (userId) => {
  try {
    const value = await onceGetUsers();
    return value[userId].favorites;
  } catch (error) {
    throw new Error ('unable to load favorites data')
  }
}

export const getFriendsFavorites = async (userId) => {
  const friends = await getFriends(userId);
  const value = await onceGetUsers();
  if(friends) {
    return friends.reduce( (accu, friendId) => {
      if(value[friendId].lastFavorited) {
        accu[value[friendId].user.firstName] = value[friendId].lastFavorited
      }
      return accu
    }, {} )  
  } else {
    return {}
  }
}

export const doWriteReviewData = async (userId, favorite, review) => {
  try {
    const { name } = favorite;
    const reviewedGame = {...favorite, review}
    const response = await db.ref('users/' + userId + '/favorites/' + name).set(reviewedGame);
    if(response.status < 300) {
      await doWriteLastFavoritedData(userId, reviewedGame)
    } else {
      throw new Error('unable to process review')
    }
  } catch (error) {
    throw new Error('unable to process review')
  }
}