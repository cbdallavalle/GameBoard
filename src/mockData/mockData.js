export const mockData = {
  mockUsers: {
    YRrtMzWo0jNquYjLCQGT8ftqz703:{
      favorites: {}, 
      friends: [], 
      lastFavorited: {}, 
      user: {
        email: "kubla@email.com",
        firstName: "kubla",
        id: "GZZGrEW9aYXzbxrdphXuq04ng6E3",
        lastName: "khan"
      }
    },
    GZZGrEW9aYXzbxrdphXuq04ng6E3:{
      favorites: {}, 
      friends: [], 
      lastFavorited: {}, 
      user: {
        email: "wren@wren.com",
        firstName: "wren",
        id: "GZZGrEW9aYXzbxrdphXuq04ng6E3",
        lastName: "little" 
      }
    }
  },
  mockAllUsersWithoutSearcher: [
    {
      email: "kubla@email.com", 
      firstName: "kubla", 
      id: "GZZGrEW9aYXzbxrdphXuq04ng6E3", 
      lastName: "khan", 
      key: "YRrtMzWo0jNquYjLCQGT8ftqz703"
    }
  ],
  mockAllUsers: [
    {
      email: "kubla@email.com", 
      firstName: "kubla", 
      id: "GZZGrEW9aYXzbxrdphXuq04ng6E3", 
      lastName: "khan", 
      key: "YRrtMzWo0jNquYjLCQGT8ftqz703"
    },
    {
      email: "wren@wren.com",
      firstName: "wren",
      id: "GZZGrEW9aYXzbxrdphXuq04ng6E3",
      lastName: "little" 
    }
  ],
  mockFriends: ['123', '456', '778'],
  mockFavorites: { 'Race for the Galaxy': {game: 'info'} },
  mockFriendsFavorites: {
    kubla: { 'Race for the Galaxy': {game: 'info'} },
    wren: { 'Captain Sonar': {game: 'info'} }
  },
  initialState: {
    firstName: '',
    lastName: '',
    email: '',
    passwordOne: '',
    passwordTwo: '',
    error: null,
    displayCreate: 'login'
  },
  mockLogInEvent: {
    target: {
      name: 'email', 
      value: 'coda@gmail.com'
    }, 
    preventDefault: () => {}
  },
  mockDefaultSearchState: {
    games: [],
    search: '',
    game: {},
    error: '',
    loading: false,
    gameAdded: 'not-added'
  },
  mockSearchEvent: {
    target: {
      name: 'search',
      value: 'Meeple Circus'
    },
    preventDefault: () => {}
  }, 
  mockSearch: [{ 
    id: '123', 
    name: 'Mysterium' 
  }],
  mockSearchGame: { 
    id: '123', 
    name: 'Mysterium', 
    thumbnail: 'src', 
    image: 'src', 
    description: 'A mystery game with pictures' 
  },
  mockMSTPUserState: {
    user: {
      uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'
    }
  },
  mockMSTPUser: {
    user: {
      uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'
    }
  }
};