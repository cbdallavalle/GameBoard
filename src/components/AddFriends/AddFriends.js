import React, { Component } from 'react';
import { db } from '../../firebase';

export class AddFriends extends Component {
  constructor(props) {
    super(props)
    this.state = {
      search: '',
      usersSearched: [],
      allUsers: []
    }
  }

  componentDidMount = async() => {
    await this.getAllUsers();
  }

  getAllUsers = async() => {
    const snapshot = await db.onceGetUsers();
    const value = await snapshot.val();
    this.setState({ allUsers: value });
  }

  handleChange = e => {
    const search = e.target.value.toLowerCase();
    const users = this.state.allUsers
    const userIds = Object.keys(users);
    const matches = userIds.filter(id => (users[id].email.toLowerCase().includes(search) || users[id].username.toLowerCase().includes(search)))
    const usersSearched = matches.map(id => users[id]);
    this.setState({usersSearched})
  }

  render() {
    return (
      <section className="AddFriends">
        <h1>Find new friend</h1>
        <input 
          type="text"
          placeholder="enter name or email" 
          onChange={this.handleChange}
        />
        <article>
          display users that match search
        </article>
      </section>
    )
  }
}

export default AddFriends