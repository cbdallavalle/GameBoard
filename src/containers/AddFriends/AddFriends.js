import React, { Component } from 'react';
import { connect } from 'react-redux';
import { db } from '../../firebase';
import { updateFriends } from '../../actions';
import addIcon from '../../assets/plus.svg';
import PropTypes from 'prop-types';
import './AddFriends.css';

export class AddFriends extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      usersSearched: [],
      allUsers: [],
      error: ''
    };
  }

  componentDidMount = async() => {
    await this.getAllUsers();
  }

  getAllUsers = async() => {
    try {
      const value = await db.onceGetUsers();
      const excludeCurrentUser = Object.keys(value)
        .filter( key => key !== this.props.user.uid);
      const allUsers = excludeCurrentUser
        .map( key =>{ return { ...value[key].user, key }; } );
      this.setState({ allUsers });
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  handleChange = event => {
    const search = event.target.value.toLowerCase();
    const users = this.state.allUsers;
    const usersSearched = users.filter(users => 
      (users.email.toLowerCase().includes(search) 
        || users.firstName.toLowerCase().includes(search) 
        || users.lastName.toLowerCase().includes(search)
      )
    );
    this.setState({ usersSearched });
  }

  displayFriends = () => {
    return this.state.usersSearched.length ?
      this.state.usersSearched.map( (friend, index) => 
        <h4 
          key={index} 
          onClick={ () => this.addFriendsToDB(friend.id) }
          id="friend-results"
        >
          <img 
            id="add-friends" 
            src={ addIcon } 
            alt="add-game" 
          />
          {friend.firstName} {friend.lastName}, {friend.email}
        </h4>
      )
      : 
      <div>
        { this.displayAllUsers() }
      </div>;
  }

  displayAllUsers = () => {
    return this.state.allUsers.map( (friend, index) => <h4 
          key={index} 
          onClick={ () => this.addFriendsToDB(friend.id) }
          id="friend-results"
        >
          <img 
            id="add-friends" 
            src={ addIcon } 
            alt="add-game" 
          />
          {friend.firstName} {friend.lastName}, {friend.email}
        </h4>)
  }

  addFriendsToDB = async (id) => {
    try {
      await db.doWriteFriendsData(this.props.user.uid, id);
      const friends = await db.getFriends(this.props.user.uid);
      this.props.updateFriends(friends);
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    return (
      <section className="AddFriends">
        <input 
          type="text"
          placeholder="enter name or email" 
          onChange={this.handleChange}
        />
        <div className="display-friends">
          <article>
            { this.displayFriends() }
          </article>
        </div>
      </section>
    );
  }
}

AddFriends.propTypes = {
  user: PropTypes.object.isRequired,
  updateFriends: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  updateFriends: friends => dispatch(updateFriends(friends))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddFriends);