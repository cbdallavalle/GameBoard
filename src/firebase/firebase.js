import * as firebase from 'firebase';

const config = {
  apiKey: "AIzaSyBDU0_Pa8xdAWKmtKKfo3QkbrtR56cl024",
  authDomain: "dallavalle-gameboard.firebaseapp.com",
  databaseURL: "https://dallavalle-gameboard.firebaseio.com",
  projectId: "dallavalle-gameboard",
  storageBucket: "",
  messagingSenderId: "250758219123"
};

if(!firebase.apps.length) {
  firebase.initializeApp(config)
}

const auth = firebase.auth();

export {
  auth,
}