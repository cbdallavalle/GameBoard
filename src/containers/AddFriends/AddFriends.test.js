/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { db } from '../../firebase';
import { AddFriends, mapStateToProps, mapDispatchToProps } from './AddFriends';
import { mockData } from '../../mockData/mockData';

describe("AddFriends", () => {
  const mockUpdateFriends = jest.fn();
  const initialState = {      
    search: '',
    usersSearched: [],
    allUsers: [],
    error: ''
  }

  it('should start with a default state', () => {
    const wrapper = shallow(
      <AddFriends 
        user={{}} 
        updateFriends={mockUpdateFriends}
      />, {disableLifeCycleMethods: true} )
    expect(wrapper.state()).toEqual(initialState);
  })

  //getAllUsers
  it('getAllUsers should update state with an error message if an error is caught', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )

    db.onceGetUsers = jest.fn().mockImplementation(() => {
      throw new Error('unable to fetch users')
    })

    await wrapper.instance().getAllUsers();

    expect(wrapper.state().error).toEqual('unable to fetch users')
  })

  it('getAllUsers should update state to have all users', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )
    
    db.onceGetUsers = () => mockData.mockUsers;
    await wrapper.instance().getAllUsers();
    expect(wrapper.state().allUsers).toEqual(mockData.mockAllUsersWithoutSearcher)
  })

  //handleChange
  it('handleChange should find the users that match the search', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )
    const mockEvent = {target:{value: 'k'}}
    await wrapper.setState({ allUsers: mockData.mockAllUsers });
    await wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().usersSearched).toEqual(mockData.mockAllUsersWithoutSearcher)
  })

  it('handleChange should return an empty array if no users match the search', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )
    const mockEvent = {target:{value: 'zzzz'}}
    await wrapper.setState({ allUsers: mockData.mockAllUsers });
    await wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().usersSearched).toEqual([])
  })

  //displayFriends
  it('displayFriends should match the snapshot if usersSearched has length', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )
    wrapper.setState({usersSearched: ['user', 'user']})
    expect(wrapper).toMatchSnapshot();
  })

  it('displayFriends should match the snpashot if usersSearched does not have length', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )
    expect(wrapper).toMatchSnapshot();
  })

  //addFriendsToDB
  it('addFriendsToDB should set state with an error if an error is caught', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )

    db.doWriteFriendsData = jest.fn().mockImplementation(() => {
      throw new Error('unable to fetch users')
    })

    db.getFriends = () => {}

    await wrapper.instance().addFriendsToDB();

    expect(wrapper.state().error).toEqual('unable to fetch users')
  })

  it('addFriendsToDB should call updateFriends with friends', async() => {
    const wrapper = await shallow(
      <AddFriends 
        user={{uid: 'GZZGrEW9aYXzbxrdphXuq04ng6E3'}} 
        updateFriends={mockUpdateFriends}
      /> )

    db.doWriteFriendsData = jest.fn();
    db.getFriends = () => mockData.mockAllUsers;
    await wrapper.instance().addFriendsToDB('1');
    expect(mockUpdateFriends).toHaveBeenCalledWith(mockData.mockAllUsers);
  })
})

describe("mapStateToProps", () => {
  it('should take in state and return a user object', () => {
    expect(mapStateToProps(mockData.mockMSTPUserState)).toEqual(mockData.mockMSTPUser)
  })
})

describe("mapDispatchToProps", () => {
  it('should call dispatch when updateFriends is called', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.updateFriends();
    expect(mockDispatch).toHaveBeenCalled();    
  })
})