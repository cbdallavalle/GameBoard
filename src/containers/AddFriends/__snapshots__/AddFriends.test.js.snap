// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFriends displayFriends should match the snapshot if usersSearched has length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFriends
    updateFriends={[Function]}
    user={
        Object {
            "uid": "GZZGrEW9aYXzbxrdphXuq04ng6E3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="enter name or email"
          type="text"
/>,
        <div
          className="display-friends"
>
          <article>
                    <h4
                              onClick={[Function]}
                    >
                               
                              , 
                    </h4>
                    <h4
                              onClick={[Function]}
                    >
                               
                              , 
                    </h4>
          </article>
</div>,
      ],
      "className": "AddFriends",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "enter name or email",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <article>
            <h4
                        onClick={[Function]}
            >
                         
                        , 
            </h4>
            <h4
                        onClick={[Function]}
            >
                         
                        , 
            </h4>
</article>,
          "className": "display-friends",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                onClick={[Function]}
>
                 
                , 
</h4>,
              <h4
                onClick={[Function]}
>
                 
                , 
</h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                  undefined,
                  ", ",
                  undefined,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                undefined,
                ", ",
                undefined,
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                  undefined,
                  ", ",
                  undefined,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                undefined,
                ", ",
                undefined,
              ],
              "type": "h4",
            },
          ],
          "type": "article",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="enter name or email"
            type="text"
/>,
          <div
            className="display-friends"
>
            <article>
                        <h4
                                    onClick={[Function]}
                        >
                                     
                                    , 
                        </h4>
                        <h4
                                    onClick={[Function]}
                        >
                                     
                                    , 
                        </h4>
            </article>
</div>,
        ],
        "className": "AddFriends",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "enter name or email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <article>
              <h4
                            onClick={[Function]}
              >
                             
                            , 
              </h4>
              <h4
                            onClick={[Function]}
              >
                             
                            , 
              </h4>
</article>,
            "className": "display-friends",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  onClick={[Function]}
>
                   
                  , 
</h4>,
                <h4
                  onClick={[Function]}
>
                   
                  , 
</h4>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                    ", ",
                    undefined,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                  ", ",
                  undefined,
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                    ", ",
                    undefined,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                  ", ",
                  undefined,
                ],
                "type": "h4",
              },
            ],
            "type": "article",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AddFriends displayFriends should match the snpashot if usersSearched does not have length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFriends
    updateFriends={[Function]}
    user={
        Object {
            "uid": "GZZGrEW9aYXzbxrdphXuq04ng6E3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="enter name or email"
          type="text"
/>,
        <div
          className="display-friends"
>
          <article>
                    <div>
                              No friends found :(
                    </div>
          </article>
</div>,
      ],
      "className": "AddFriends",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "enter name or email",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <article>
            <div>
                        No friends found :(
            </div>
</article>,
          "className": "display-friends",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              No friends found :(
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No friends found :(",
            },
            "ref": null,
            "rendered": "No friends found :(",
            "type": "div",
          },
          "type": "article",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="enter name or email"
            type="text"
/>,
          <div
            className="display-friends"
>
            <article>
                        <div>
                                    No friends found :(
                        </div>
            </article>
</div>,
        ],
        "className": "AddFriends",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "enter name or email",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <article>
              <div>
                            No friends found :(
              </div>
</article>,
            "className": "display-friends",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                No friends found :(
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No friends found :(",
              },
              "ref": null,
              "rendered": "No friends found :(",
              "type": "div",
            },
            "type": "article",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
